src/Uno.UWP/Globalization/DateTimeFormatting/DateTimeFormatter.cs

```diff
@@ -797,27 +797,68 @@ void AddToBuilder(StringBuilder builder, char lastChar, int count)
 
 		string ConstructPattern(string str)
 		{
+			if (string.IsNullOrEmpty(str))
+			{
+				return string.Empty;
+			}
+
+			char singleQuoteChar = '\'';
 			var builder = new StringBuilder();
-			char lastChar = str[0];
-			int count = 1;
-			for (int i = 1; i < str.Length; i++)
+			int i = 0;
+			int len = str.Length;
+
+			while (i < len)
 			{
-				if (lastChar != str[i])
+				char currentChar = str[i];
+
+				// Handle quoted section (literals) "MMMM 'de' yyyy"
+				if (currentChar == singleQuoteChar)
 				{
-					AddToBuilder(builder, lastChar, count);
+					i++;
+					var literal = new StringBuilder();
 
-					lastChar = str[i];
-					count = 1;
+					while (i < len)
+					{
+						if (str[i] == singleQuoteChar)
+						{
+							if (i + 1 < len && str[i + 1] == singleQuoteChar)
+							{
+								literal.Append(singleQuoteChar);
+								i += 2;
+							}
+							else
+							{
+								// End of quoted section
+								i++;
+								break;
+							}
+						}
+						else
+						{
+							literal.Append(str[i]);
+							i++;
+						}
+					}
+
+					builder.Append(literal);
 				}
 				else
 				{
-					count++;
+					char lastChar = currentChar;
+					int count = 1;
+					i++;
+
+					while (i < len && str[i] == lastChar && str[i] != '\'')
```

Comment:
we are within the else block of
```cs
if (currentChar == singleQuoteChar) // L822
```
there is no need to check `str[i] is not SingleQuoteChar`, since you just confirmed `str[i] == lastChar/*aka currentChar*/` which cant be single quote